openlibrary/templates/work_search.html

```diff
@@ -168,20 +178,16 @@ <h1>$_("Search")</h1>
         <div class="resultsContainer search-results-container">
         <div id="searchResults">
           <ul class="list-books">
-            $ works = add_availability([get_doc(d) for d in docs])
+            $ works = [get_doc(d) for d in docs]
+            $ add_availability([(w.get('editions') or [None])[0] or w for w in works])
             $ username = ctx.user and ctx.user.key.split('/')[-1]
             $if username:
                 $ works = add_read_statuses(username, works)
 
             $for work in works:
-                $ ocaid = work.ia[0] if work.ia else None
-                $ availability = (work.get('availability') or {}).get('status')
-                $# if we're explicitly showing *everything*...
-                $# or if we're showing only things with ocaids which are available...
-                $if 'has_fulltext' not in param or (ocaid and availability and availability not in ['error', 'private']):
-                    $ read_status = work.get('readinglog', None)
-                    $ dropper = macros.ReadingLogDropper([], work=work, reading_log_only=True, page_url="/search", users_work_read_status=read_status)
-                    $:macros.SearchResultsWork(work, reading_log=dropper)
+                $ read_status = work.get('readinglog', None)
+                $ dropper = macros.ReadingLogDropper([], work=work, reading_log_only=True, page_url="/search", users_work_read_status=read_status)
+                $:macros.SearchResultsWork(work, reading_log=dropper)
```

Comment:
We already check the permissions above, let's send that in instead of recomputing at each search card

Reply:
Done